src/WebJobs.Script.WebHost/Security/HostSecrets.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class HostSecrets
+    {
+        [JsonProperty(PropertyName = "masterKey")]
+        public Key MasterKey { get; set; }
+
```

Comment:
Perhaps not - the master key unlike all the function keys is never supposed to be distributed to apps, web hooks, etc. So the scenarios calling for it to be rolled are much less. Since the masterKey is really only ever used by the app owner in admin scenarios, rolling it is as simple as changing its value.